### ISSUE TITLE
new handling for site map image

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#471]
site map images need the new image handling.

**Comment by urapadmin** [comment]
### @test
Site map images
 
1. add a new site map (not through the gallery) in the tab (instead of in the popup with the add from gallery button)
2. make sure that after adding the image it can be found in the gallery when searching for the site's id
3. do the same thing but this time in the popup
4. add another image from the gallery and make sure afterwards that it can be found in the gallery under the site id (be careful to use an image that didn't show up for this site to begin with)
5. delete the images (there is a remove image button in the popup)
6. make sure that they are not any longer found in the gallery under the site id

**Comment by urapadmin** [comment]
see also https://www.meritaten.net/urapdev/doku.php?id=how_to_test_an_image_container, but this here is simpler because there is only once image.

**Comment by SamKimball-BrownU** [comment]
Oof, using any method, it is unsearchable in gallery (by site id or site title). Deleting it had no effect, as it wasn't there to begin with (I tried it anyway just to see though)

Version 8.9.9

**Comment by urapadmin** [comment]
### v8.9.10:
I cannot reproduce this entirely. It did indeed use only either the short or the long site name (can't remember), so it was found only under one of the terms. But it was added to the cache, so I don't know what is different on your site. The new version is improved in that it adds both, the site id and the fully qualified name to the cache. 
What did not work here was adding an image as a site map and then deleting it immediately via backspace. That would have left it in the cache - so it would have been found. 

Please try with the new version. If the error remains, here are some questions:
Where did you add the image? In the container next to site notes or in the container behind the site configuration (the button with the gear icon on the start page)? Through the gallery or by adding a new file? Does anything make a difference at all?

**Comment by SamKimball-BrownU** [comment]
USTP 1.9.11/Windows 10

Using gallery to add images seems to resolve issue, but that is the only thing I could find that worked.

**Here are scenarios that I tried that did not work:**
Make new site, exit out, go back to site (gear icon) and add it not using the popup
Make new site, exit out, go back to site (gear icon) and use popup
Make new site, exit out, go back to site (clipboard icon) and do not use popup
Make new site, exit out, go back to site (clipboard icon) and use popup
Make new site, add site map there not using popup
Make new site, add site map there using popup



**Here are scenarios I tried that DID WORK (at least partially)**
Make new site, exit out, go back to site (gear icon) and use gallery 
        However, deleting it using the trash can (not in popup)  does not remove it from searches
        Other ways of deleting work as intended
Make new site, exit out, go back to site (clipboard) and use gallery 
         All methods of deleting it in the clipboard icon menu works as intended
Make new site, add image there using the gallery
         Deleting is as before, you have to choose which menu you go into and then it operates like 
             above

**In summary:** Adding images using gallery works for the most part, except for issue with deleting using the trashcan under gear icon (not in popup), see above. All other methods of adding images appear visually, but not when searched in gallery.

**Comment by SamKimball-BrownU** [comment]
Also, I have a suspicion that the issue 1 in #472 is related to this one.

**Comment by urapadmin** [comment]
the only thing I was able to reproduce here is the one with the trash can. I singled that issue out to thise ticket:

**Comment by urapadmin** [comment]
I suspect a connection to #472, too. So I look deeper into this here as soon as I get the database that shows the behaviour from you (see #472).

**Comment by urapadmin** [comment]
have I asked you on what filemaker version you are?

**Comment by SamKimball-BrownU** [comment]
I'm on filemaker pro 18 advanced, version number 18.0.3.317

**Comment by urapadmin** [comment]
that is the same version I have. So that's not the issue.
Ok, one last try before I have to prepare a debugging version:
Let's stick with the simple site map example. Here is exactly what I do:

- Gallery for POP is empty to begin with:
![image](https://user-images.githubusercontent.com/38838314/79132116-b9341580-7d77-11ea-81a7-22439c0b739e.png)
- I am signed in as admin, since that is what you must have been using, too.

1. I enter site POP with the gear icon
2. Under site map I use the context menu and Insert Picture... to select and insert an image
3. Image is there, I leave.
4. back in the gallery I have to reenter "POP" or select "site" again to make the screen refresh (that is a bit of a bug in and of itself).
5. There is my image and this is the information it gives me:
![image](https://user-images.githubusercontent.com/38838314/79132686-c43b7580-7d78-11ea-9952-8fd9b52ac175.png)

Please try to follow those exact same steps. Anything different?

**Comment by SamKimball-BrownU** [comment]
AHA!!!!! So, when I right click and do "insert picture" it is exactly as you describe. But, when I drag and drop an image file onto it, it works as I have described! I didn't even realize there was an option for right clicking and choosing "insert picture." I guess this adds another scenario to test for in the future. I will do these tests on the other relevant tickets after classes today. Glad that it's finally reproducible (at least on my end)

**Comment by urapadmin** [comment]
that was good testing! Filemaker processes drag and drop unlike any other operation. Frustrating. However, not only the code but also testing gets more complicated: https://www.meritaten.net/urapdev/doku.php?id=how_to_test_an_image_container#a_test_schema.

**Comment by urapadmin** [comment]
### fixed in 8.9.12, 
... I hope

**Comment by SamKimball-BrownU** [comment]
INSERTING:

On the dragging and dropping issue: It does pop up with a warning, but it took a while for me to actually see it because of my testing methodology. Basically, I sometimes tested the drag and drop operation right after testing some other method, so then the popup wouldn't show (as you describe in the new wiki entry). So, it seems to be working as intended with the popup, but with the known quirks of filemaker handling of drag & drop operations 

Whenever the error popup shows, it adds the image that was dragged and dropped perfectly, being searchable and such. Whenever the error popup didn't show, it does not have any identifier attached. I found this fairly hard to replicate, however.

On insert in popup/not using context menu (right click): There is the same window popup with the same errors as in #529 . However, after clicking revert a couple times, it works as intended. 

Inserting from Gallery: works 

DELETING: Does not work by any method I found. They still show up under the site ID. Issue in #524 is now not limited to just the trashcan, but by other methods too.

**Comment by urapadmin** [comment]
Hm, I cannot reproduce that even with the database I got from you. Here I have one idea: Is it possible that you stumble over the little bug that the gallery does not refresh the list when you reenter it? So "Sam's Site" and "SITE" are still selected as filters but the results you see are still the results from the last search. If you just select "SITE" again the results are refreshed.

**Comment by SamKimball-BrownU** [comment]
The issue persists when logged in with sgk, even when you look in testing menu

When logged in with admin, the issue disappears


(there's certainly a pattern here at least....)

**Comment by urapadmin** [comment]
yes! Indeed, I can reproduce it when logged in as a user without any privileges. That will lead me to the problem.... but first some sleep...

**Comment by urapadmin** [comment]
### v8.9.14: all issues here should be solved now

**Comment by SamKimball-BrownU** [comment]
RESOLVED!!! (Thank goodness)

**Comment by urapadmin** [comment]
Yippieh.

**Comment by urapadmin** [comment]
reopened because image proxy was not set correctly.

**Comment by urapadmin** [comment]
### v9.1
will this ever end?
